server/lyft/gateway/autoplan_builder.go

```diff
@@ -0,0 +1,143 @@
+	"github.com/runatlantis/atlantis/server/events"
+	"github.com/runatlantis/atlantis/server/events/command"
+	"github.com/runatlantis/atlantis/server/events/metrics"
+	"github.com/runatlantis/atlantis/server/events/models"
+	"github.com/runatlantis/atlantis/server/events/vcs"
+	"github.com/runatlantis/atlantis/server/logging"
+	"github.com/runatlantis/atlantis/server/recovery"
+	"github.com/uber-go/tally"
+	"strconv"
+)
+
+// AutoplanValidatorBuilder handles setting up repo cloning and checking to verify of any terraform files have changed
+type AutoplanValidatorBuilder struct {
+	Logger                        logging.SimpleLogging
+	Scope                         tally.Scope
+	VCSClient                     vcs.Client
+	PreWorkflowHooksCommandRunner events.PreWorkflowHooksCommandRunner
+	DisableAutoplan               bool
+	Drainer                       *events.Drainer
+	GlobalCfg                     valid.GlobalCfg
+	// AllowForkPRs controls whether we operate on pull requests from forks.
+	AllowForkPRs bool
+	// AllowForkPRsFlag is the name of the flag that controls fork PR's. We use
+	// this in our error message back to the user on a forked PR so they know
+	// how to enable this functionality.
+	AllowForkPRsFlag string
+	// SilenceForkPRErrors controls whether to comment on Fork PRs when AllowForkPRs = False
+	SilenceForkPRErrors bool
+	// SilenceForkPRErrorsFlag is the name of the flag that controls fork PR's. We use
+	// this in our error message back to the user on a forked PR so they know
+	// how to disable error comment
+	SilenceForkPRErrorsFlag string
+	// SilenceVCSStatusNoPlans is whether autoplan should set commit status if no plans
+	// are found
+	silenceVCSStatusNoPlans bool
+	// SilenceVCSStatusNoPlans is whether any plan should set commit status if no projects
+	// are found
+	silenceVCSStatusNoProjects bool
+	CommitStatusUpdater        events.CommitStatusUpdater
+	PrjCmdBuilder              events.ProjectPlanCommandBuilder
+	PullUpdater                *events.PullUpdater
+}
+
+func (r *AutoplanValidatorBuilder) IsValid(baseRepo models.Repo, headRepo models.Repo, pull models.PullRequest, user models.User) bool {
+	if opStarted := r.Drainer.StartOp(); !opStarted {
+		r.Logger.Err("Atlantis is shutting down, cannot process current event")
+		r.Scope.Counter(metrics.ExecutionErrorMetric).Inc(1)
+		return false
+	}
+	defer r.Drainer.OpDone()
+
+	log := r.Logger.WithHistory(
+		"repository", baseRepo.FullName,
+		"pull-num", strconv.Itoa(pull.Num),
+	)
+	defer r.logPanics(log)
+
+	timer := r.Scope.Timer(metrics.ExecutionTimeMetric).Start()
+	defer timer.Stop()
+
+	ctx := &command.Context{
+		User:     user,
+		Log:      log,
+		Scope:    r.Scope,
+		Pull:     pull,
+		HeadRepo: headRepo,
+		Trigger:  command.AutoTrigger,
+	}
+	if !r.validateCtxAndComment(ctx) {
+		r.Scope.Counter(metrics.ExecutionErrorMetric).Inc(1)
+		return false
+	}
+	if r.DisableAutoplan {
+		r.Scope.Counter(metrics.ExecutionErrorMetric).Inc(1)
```

Comment:
This is weird to me, it shouldn't be an error because a flag is set. Let's remove this, we don't even use this code path anyways.